superset/viz.py

```diff
@@ -787,8 +787,8 @@ def query_obj(self):
         return d
 
     def get_data(self, df):
-        df['x'] = df[[self.x_metric]]
-        df['y'] = df[[self.y_metric]]
+        df['x'] = df[[self.x_metric]].ix[:, 0]
+        df['y'] = df[[self.y_metric]].ix[:, 0]
```

Comment:
I've written a fair amount of pandas and I'm unclear what this does... Should we just raise above in `query_obj` when `self.x_metric == self.y_metric` ?

Reply:
This is the meaning.